### PR TITLE
WIP:DependenciesViewer-DependenciesDataGeneration into master

### PR DESCRIPTION
**Pull request by heikob2**
Purpose:
- enable dependencies-data-generation on ci/teamcity and send the data to a remote server
- can be executed after build and before publish
- only on ci and when system-property "system.ddgEnabledCi" = "true" is set at the teamcity-configurations

Changes: 
- added folder "./dv-ddg-client"
- adapted "**build.gradle**":
   - added code for dependencies-data-generation at the bottom
   - dependencies-data-generation dependsOn "build_languages"
   - added dependency for "publish"-gradle-task: dependsOn buildWithDdg

The "build.gradle"-file needs special attention.

### CODE REVIEW

**Comment by heikob2** [comment]
implementation is done. @slisson could you please review?

File: dv-ddg-client/ddgconfig.gradle [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+//relative to iets3.opensoure/build.gradle
+//seems not be applicable on ci-server
+project.ext.projects_home = './..'
+//not used for iets3
+project.ext.user_home = '-'
+
+project.ext.plugins_location = "${mps_home}/plugins"
+//only used for run_mbeddr_core
+project.ext.application_support_plugins_location = "${user_home}/Library/Application Support/MPS2018.3"
```
Comment (slisson) [inline dv-ddg-client/ddgconfig.gradle]: There is a variable `ext.mpsPluginsDir` in the main build script.

**Comment by slisson** [comment]
Looks like this doesn't add any features to the MPS-Extensions project, but just executes a dependency analysis on the artifacts. If this is true, changes to this repository are not necessary and this could be done in a less invasive way (download the artifacts from the nexus repository as all other projects do).

**Comment by heikob2** [comment]
Yes it does not add any new features to mps-extension. 

For the analysis, the dependencies-generator opens the mps-extensions-project (".mps"-folder) in a headless-mps-environment and calculates the dependencies for the MPS-Project.

It not uses the generated artifacts. (from the nexus repository)

**Comment by slisson** [comment]
But do we need to modify each repository that we want to analyze (and copy all these jars to it)? Or can we have a separate project that has a dependency on all the projects (git submodules?) it wants to analyze.

**Comment by heikob2** [comment]
This approach sounds to be the better option. Thank you.

**Comment by kbirken** [comment]
I agree with the remark that the change is quite invasive. It should at least be possible to deploy all the jars on Nexus and just do some additions to the project's gradle-build. The gradle-task can then download the jars (on demand) and do the dependency analysis. How about this approach?

**Comment by heikob2** [comment]
Yes. This is also possible. 👍

**Pull request closed without merging**